packages/core/src/components/TextField/TextField.jsx

```diff
@@ -119,7 +124,7 @@ export class TextField extends React.PureComponent {
         className={fieldClasses}
         id={this.id}
         /* eslint-disable no-return-assign */
-        ref={focusTrigger ? loader => (this.loader = loader) : fieldRef}
+        ref={focusTrigger ? ref => (this.focusRef = ref) : inputRef || fieldRef}
```

Comment:
Whatever is done to the address the readability of this property in `Dropdown` could be done here as well